### ISSUE TITLE
"_on_navigation_agent_3d_velocity_computed" signal never being emitted

### ISSUE THREAD

**Issue by JGGentzel** [godotengine/godot#58578]
### Godot version

4.3 alpha

### System information

macOS Monterey version 12.2 MacBook Pro (16-inch, 2019) Processor: 2.6 GHz 6-Core Intel Core i7 Memory: 16 GB 2667 MHz DDR4  Graphics: Intel UHD Graphics 630 1536 MB 

### Issue description

When I use the set_velocity() method on the NavMeshAgent3D the   "_on_navigation_agent_3d_velocity_computed" signal is never emitted. 

### Steps to reproduce

Used this video as a general guide:
https://www.youtube.com/watch?v=1c26NezEhmQ

A path is being computed and I am able to set a motion_velocity on the CharacterBody3D based off the get_next_location() method on the NavMeshAgent3D. When I use the set_velocity() method on the NavMeshAgent3D the   "_on_navigation_agent_3d_velocity_computed" signal is never emitted. 

### Minimal reproduction project

[Archive.zip](https://github.com/godotengine/godot/files/8147820/Archive.zip)

**Comment by DarkKilauea** [comment]
This is working fine in my test project with the latest master branch (and has been working for at least the last month): https://github.com/DarkKilauea/godot-nav-link-demo/blob/master/character.gd

Double check your setup, it can be a little tricky to get right.

**Comment by JGGentzel** [comment]
Yep. Your example works. I changes mine to more closely match yours and it works now. Thank you.

My issue is resolved.
